### DISCUSSION TITLE
Add support LITER and GALLON volume unit for GAS

### DISCUSSION THREAD

**Discussion by Laxilef** [home-assistant/architecture#994] [Entity Models]
### Problem
At the moment, "device_class: gas" entities can be in m³, ft³, CCF, but there is also liquefied gas, which is usually measured in liters.
When converting liquefied gas from liters to cubic meters, we get a not entirely correct display on the Energy panel:

<img src="https://community-assets.home-assistant.io/original/4X/b/9/f/b9f20b5de86808de79bfbcbb3850ce557bf14ab0.png">

Liquefied gas is used in gas cylinders and gasholders. It is used for gas stoves and home heating with a gas boiler in the same way as regular gas. But due to the fact that it is in a liquid state, it is usually measured in liters. It is not correct to convert it to m3.
You can open any instructions for a gas boiler and see the consumption in liters for liquefied gas.

I have a small telemetry project for a gas holder, where this would be relevant:
https://github.com/Laxilef/GasholderMon/blob/9828bc4305d8a054a5639aa859cadb5aaf4dd951/src/HomeAssistantHelper.h#L342C1-L344C43

[On the forum](https://community.home-assistant.io/t/energy-dashboard-support-gas-in-liters-unit/521643), some people also wrote that this was important for them.

### Proposal
Add the ability to use liters and gallons for "device_class: gas" entities.
PR: https://github.com/home-assistant/core/pull/102596

**Comment by Laxilef** [comment]
I'll give an example.
Since liquefied gas is measured in liters, the price for it is also set in liters.

Let's say we want to track liquefied gas consumption and its cost on the Energy panel.

Currently it is not possible to create a gas sensor in liters, so it is necessary to convert it to cubic meters.
However, 1000 liters of liquefied gas is not equal to 1 cubic meter. Therefore, for conversion it is necessary to know the ratio of propane and butane in the gas. In practice, it is impossible to know this, because the proportions are different in winter and summer. And gas suppliers never report exact proportions.

Take the average value: 1 liter of liquefied gas = 0.24 m3.
Now we need to create a "template sensor" to calculate in cubic meters: liters * 0.24

But to calculate the cost, it is necessary to convert the price, because when purchasing, the price is per liter. And you need to create another “template sensor”: price per liter / 0.24.

All this introduces additional difficulties and inaccuracies, because the coefficient is not exact and depends on many factors.

This is why I propose adding this feature.

**Comment by balloob** [comment]
We are unable to accept this because every gas requires a different conversion ratio. Home Assistant only works with fixed conversion rates. If you want to use for pricing, you should convert your price instead to m3.

**Reply by Laxilef** [reply]
Can't we prohibit the conversion from liters to cubic meters for gas?
That is, we can leave the conversion of liters to gallons, and cubic meters to cubic feet. But not liters to cubic meters and not gallons to cubic feet.
